dtls.c

```diff
@@ -598,26 +592,12 @@ int janus_dtls_srtp_create_sctp(janus_dtls_srtp *dtls) {
 		return -4;
 	if(janus_flags_is_set(&handle->webrtc_flags, JANUS_ICE_HANDLE_WEBRTC_ALERT))
 		return -5;
-	dtls->sctp = janus_sctp_association_create(dtls, handle->handle_id, 5000);
+	dtls->sctp = janus_sctp_association_create(dtls, handle, 5000);
 	if(dtls->sctp == NULL) {
 		JANUS_LOG(LOG_ERR, "[%"SCNu64"] Error creating SCTP association...\n", handle->handle_id);
 		return -6;
 	}
-	/* We need to start it in a thread, since it has blocking accept/connect stuff */
-	janus_refcount_increase(&dtls->ref);
-	janus_refcount_increase(&dtls->sctp->ref);
-	GError *error = NULL;
-	char tname[16];
-	g_snprintf(tname, sizeof(tname), "sctpinit %"SCNu64, handle->handle_id);
-	g_thread_try_new(tname, janus_dtls_sctp_setup_thread, dtls, &error);
-	if(error != NULL) {
-		/* Something went wrong... */
-		janus_refcount_decrease(&dtls->ref);
-		janus_refcount_decrease(&dtls->sctp->ref);
-		JANUS_LOG(LOG_ERR, "[%"SCNu64"] Got error %d (%s) trying to launch the DTLS-SCTP thread...\n", handle->handle_id, error->code, error->message ? error->message : "??");
-		return -7;
-	}
-	return 0;
+	return janus_sctp_association_setup(dtls->sctp);
```

Comment:
I'm wondering if we should merge `janus_sctp_association_setup` in `janus_sctp_association_create`, since there's no separate thread needed to call that method anymore.

Reply:
Done